sympy/stats/crv_types.py

```diff
@@ -134,7 +134,8 @@ def ContinuousRV(symbol, density, set=Interval(-oo, oo)):
     >>> P(X>0)
     1/2
     """
-    pdf = Lambda(symbol, density)
+    pdf = Piecewise((density, set.as_relational(symbol)), (0, True))
```

Comment:
did you check if `set.as_relational` is stable? Does it ever raise exceptions?

Reply:
I checked it for intervals and unions of intervals, and it did not raise any issues. I think most sets passed to `ContinuousRV` would be of these type.l 
`as_relational` does not work with sets such as `S.Integers`, or `Range`, and so `DiscreteRV` stil has the issue.